Signal/src/ViewControllers/MessageDetailViewController.swift

```diff
@@ -212,8 +213,15 @@ class MessageDetailViewController: OWSViewController, MediaGalleryDataSourceDele
                     groupRows.append(divider)
                 }
 
-                for recipientId in thread.recipientIdentifiers {
-                    let (recipientStatus, shortStatusMessage, _) = MessageRecipientStatusUtils.recipientStatusAndStatusMessage(outgoingMessage: outgoingMessage, recipientId: recipientId, referenceView: self.view)
+                let messageRecipientIds = outgoingMessage.recipientIds()
+
+                for recipientId in messageRecipientIds {
```

Comment:
Nit:

how about

```
for (recipientId, recipientState) in ougoingMessage.recipientStateMap {

}
```

So we can avoid the unnecessary `recipientStateMap -> [recipientIds] -> [RecipientState]` roundtrip and (we'd also obviate the `guard`)

Reply:
I'd like to leave it this way so that `recipientStateMap` can remain a private implementation detail of `TSOutgoingMessage`.  `TSOutgoingMessageRecipientState` are mutable, and by ensuring that we only "leak" copies, we can ensure that only this "owning" instance will mutate its recipient states.